src/core/data.jl

```diff
@@ -7,10 +7,31 @@ function update_data!(data::Dict{String,<:Any}, new_data::Dict{String,<:Any})
     else
         Memento.warn(_LOGGER, "running update_data with data that does not include per_unit field, units may be incorrect")
     end
+
     _update_data!(data, new_data)
 end
 
 
+function modify_data_with_function!(data::Dict{String,<:Any}, it::String, modification_function::Function; apply_to_nws::Bool = true)
```

Comment:
I agree with everything said here, and renaming this `apply!` is a much better idea. I'll change the flag to `is_multinetwork_function` (since it's a Boolean and `multinetwork_function` might be a misleading name for the parameter type).